core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala

```diff
@@ -105,7 +105,11 @@ class NettyBlockRpcServer(
         val blockId = BlockId(uploadBlock.blockId)
         logDebug(s"Receiving replicated block $blockId with level ${level} " +
           s"from ${client.getSocketAddress}")
-        blockManager.putBlockData(blockId, data, level, classTag)
+        val blockStored = blockManager.putBlockData(blockId, data, level, classTag)
+        if (!blockStored) {
+          throw new Exception(s"Upload block for $blockId failed. This " +
+            s"mostly happens when there is not sufficient space available to store the block.")
+        }
```

Comment:
How about replying the client with `responseContext.ononFailure(e)` since this is a controlled exception?

Reply:
Yes I initially used responseContext.ononFailure(e) only. The issue is that responseContext variable is not available in the streaming flow https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L669 . Also passing responseContext variable till that point will need interface changes in BlockDataManager#putBlockDataAsStream interface method. So we have to use exception in that flow.  To make things consistent at both the places, I used exception here also. @Ngone51  Should we use responseContext.onFailure instead of exception here?

Reply:
> Should we use responseContext.onFailure instead of exception here?

Yeah. I'd prefer to only use it here. But I don't expect to use it in `putBlockDataAsStream`.

Reply:
Seems the common pattern in the code base is to call `responseContext.onFailure` when in the `receive` method.

Reply:
done.